cupy/_creation/from_data.py

```diff
@@ -164,13 +174,67 @@ def fromfile(*args, **kwargs):
     return asarray(numpy.fromfile(*args, **kwargs))
 
 
-# TODO(okuta): Implement fromfunction
+def fromfunction(*args, **kwargs):
+    """Construct an array by executing a function over each coordinate.
 
+    .. note::
+        Uses NumPy's ``fromfunction`` and coerces the result to a CuPy array.
+
+    .. seealso:: :func:`numpy.fromfunction`
+    """
+    return asarray(numpy.fromfunction(*args, **kwargs))
 
-# TODO(okuta): Implement fromiter
 
+def fromiter(*args, **kwargs):
+    """Create a new 1-dimensional array from an iterable object.
 
-# TODO(okuta): Implement fromstring
+    .. note::
+        Uses NumPy's ``fromiter`` and coerces the result to a CuPy array.
+
+    .. seealso:: :func:`numpy.fromiter`
+    """
+    return asarray(numpy.fromiter(*args, **kwargs))
+
+
+def fromstring(*args, **kwargs):
+    """A new 1-D array initialized from text data in a string.
+
+    .. note::
+        Uses NumPy's ``fromstring`` and coerces the result to a CuPy array.
+
+    .. seealso:: :func:`numpy.fromstring`
+    """
+    return asarray(numpy.fromstring(*args, **kwargs))
 
 
-# TODO(okuta): Implement loadtxt
+def fromregex(*args, **kwargs):
```

Comment:
I'm sorry but `fromregex` cannot be implemented in CuPy as structured arrays are not supported yet in CuPy :bow:
Could you remove this one?

Reply:
Sure, I think I have accidently implemented this one since there were a TODO there :sweat_smile: